homeassistant/components/neato/__init__.py

```diff
@@ -216,11 +134,18 @@ def login(self):
             self.my_neato = self._neato(
                 self.config[CONF_USERNAME], self.config[CONF_PASSWORD], self._vendor
             )
-            return True
-        except HTTPError:
+            self.logged_in = True
+        except (HTTPError, ConnError):
             _LOGGER.error("Unable to connect to Neato API")
+            self.logged_in = False
             return False
```

Comment:
If we don't check the return value we can just `return` here. Or remove use of `self.logged_in` and check the return value instead.

Reply:
yea, you are right